packages/backend/src/shared/utils.ts

```diff
@@ -171,6 +172,64 @@ async function canApiSignatureBeProcessed(
   return true
 }
 
+/*
+  Verifies http signatures by first attempting to replicate it with a secret
+  associated with a tenant id in the headers, then with the configured admin secret.
+
+  If a tenant secret can replicate the signature, the request is tenanted to that particular tenant.
+  If the environment admin secret replicates the signature, then it is an operator request with elevated permissions.
+  If neither can replicate the signature then it is unauthorized.
+*/
+export async function verifyTenantOrOperatorApiSignature(
+  ctx: TenantedHttpSigContext,
+  config: IAppConfig
+): Promise<boolean> {
+  ctx.tenant = undefined
+  ctx.isOperator = false
+  const { headers } = ctx.request
+  const signature = headers['signature']
+  if (!signature) {
+    return false
+  }
+
+  const tenantService = await ctx.container.use('tenantService')
+  const tenantId = headers['tenantid']
+  const tenant = tenantId ? await tenantService.get(tenantId) : undefined
+
+  if (!(await canApiSignatureBeProcessed(signature as string, ctx, config)))
+    return false
+
+  // First, try validating with the tenant api secret
+  if (
+    tenant?.apiSecret &&
+    verifyApiSignatureDigest(
+      signature as string,
+      ctx.request,
+      config,
+      tenant.apiSecret
+    )
+  ) {
+    ctx.tenant = tenant
+    return true
+  }
+
+  // Fall back on validating with operator api secret if prior validation fails
+  if (
+    verifyApiSignatureDigest(
+      signature as string,
+      ctx.request,
+      config,
+      config.adminApiSecret as string
+    )
+  ) {
+    ctx.tenant = tenant
+    ctx.isOperator = true
+    return true
+  }
```

Comment:
(related to my [PR](https://github.com/interledger/rafiki/pull/3156/files#diff-b3971e30f50aa3c5a4843015630a6aafd3ea2f1846a23763dc7189d7ce80c462))
If we think that `apiSecret` is required in the DB schema, and we are seeding the operator tenant, we could actually avoid this extra check, and just determine whether its the operator by doing `tenant.apiSecret = config.adminApiSecret` after verifyingApiSignatureDigest